### PR TITLE
add ‘suspended’ field to the user object

### LINKED ISSUE #315: Add `suspend` field to the user and the UI to allow suspending of a user account

### PR DESCRIPTION
**Pull request by darkdarkdragon**
**What does this pull request do? Explain your changes. (required)**
Add ability to suspend user account.

**Specific updates (required)**
- Add `suspended` field to the user object
- Add admin's UI to change it

**Does this pull request close any open issues?**
Fixes #315 

**Screenshots (optional):**
image

image

### CODE REVIEW

File: packages/api/src/controllers/stream.js [hunk 1]
```diff
@@ -654,6 +654,17 @@ app.put("/:id/setactive", authMiddleware({}), async (req, res) => {
     return res.json({ errors: ["not found"] });
   }
 
+  const user = await req.store.get(`user/${stream.userId}`);
```
Comment (iameli) [inline packages/api/src/controllers/stream.js]: Can we do `db.user.get()`?

File: packages/www/hooks/use-api.tsx [hunk 2]
```diff
@@ -309,6 +309,33 @@ const makeContext = (state: ApiState, setState) => {
...
+    ): Promise<[Response, ApiError]> {
+      const [res, body] = await context.fetch(`/user/${userId}/suspended`, {
+        method: "PATCH",
+        body: JSON.stringify({ suspended }),
+        headers: {
+          "content-type": "application/json",
+        },
+      });
+
+      if (res.status !== 204) {
```
Comment (iameli) [inline packages/www/hooks/use-api.tsx:324]: Redundant with line 327, can't hit both code paths

File: packages/www/hooks/use-api.tsx [hunk 3]
```diff
@@ -309,6 +309,33 @@ const makeContext = (state: ApiState, setState) => {
       return res;
     },
 
+    async setSuspended(
```
Comment (iameli) [inline packages/www/hooks/use-api.tsx]: Perhaps `setUserSuspended`?

File: packages/api/src/controllers/stream.js [hunk 4]
```diff
@@ -954,6 +965,18 @@ app.post("/hook", async (req, res) => {
     res.status(404);
     return res.json({ errors: ["not found"] });
   }
+
+  const user = await req.store.get(`user/${stream.userId}`);
```
Comment (iameli) [inline packages/api/src/controllers/stream.js]: db.user.get

**Review by iameli — CHANGES_REQUESTED** [review]
Minor changes, feel free to merge after.

File: packages/api/src/controllers/stream.js (hunk 4 above)
Comment (darkdarkdragon) [reply packages/api/src/controllers/stream.js]: fixed

File: packages/www/hooks/use-api.tsx (hunk 3 above)
Comment (darkdarkdragon) [reply packages/www/hooks/use-api.tsx]: fixed

File: packages/www/hooks/use-api.tsx (hunk 2 above)
Comment (darkdarkdragon) [reply packages/www/hooks/use-api.tsx:324]: fixed

File: packages/api/src/controllers/stream.js (hunk 1 above)
Comment (darkdarkdragon) [reply packages/api/src/controllers/stream.js]: fixed

**Pull request merged**